engine/apps/labels/tasks.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import typing
+
+from celery.utils.log import get_task_logger
+from django.conf import settings
+from django.utils import timezone
+
+from apps.labels.client import LabelsAPIClient
+from apps.labels.utils import LABEL_OUTDATED_TIMEOUT_MINUTES, LabelKeyData, LabelsData, get_associating_label_model
+from apps.user_management.models import Organization
+from common.custom_celery_tasks import shared_dedicated_queue_retry_task
+
+logger = get_task_logger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+@shared_dedicated_queue_retry_task(
+    autoretry_for=(Exception,), retry_backoff=True, max_retries=1 if settings.DEBUG else None
+)
+def update_labels_cache_for_key(label_data: "LabelKeyData"):
```

Comment:
Not really sure there is a way, but wondering if it would be possible to refactor these `update_labels_*` tasks into a more general one?

Reply:
We are working with different types of data there (`LabelsData: list` and `LabelKeyData: dict`), but probably it makes sense to unify it somehow 🤔